mm2src/coins/z_coin.rs

```diff
@@ -1023,27 +995,50 @@ impl<'a> ZCoinBuilder<'a> {
             // This is not used for Zcoin so we just provide a default value
             path_to_address: HDPathAccountToAddressId::default(),
         };
-        ZCoinBuilder {
+
+        let z_spending_key = match z_spending_key {
+            Some(ref z_spending_key) => z_spending_key.clone(),
+            None => extended_spending_key_from_protocol_info_and_policy(
+                &protocol_info,
+                &priv_key_policy,
+                z_coin_params.account,
+            )?,
+        };
+
+        let (_, my_z_addr) = z_spending_key
+            .default_address()
+            .map_err(|_| MmError::new(ZCoinBuildError::GetAddressError))?;
+
+        let my_z_addr_encoded =
+            encode_payment_address(protocol_info.consensus_params.hrp_sapling_payment_address(), &my_z_addr);
+
+        Ok(ZCoinBuilder {
             ctx,
             ticker,
             conf,
             z_coin_params,
             utxo_params,
             priv_key_policy,
-            db_dir_path,
             z_spending_key,
+            my_z_addr,
+            my_z_addr_encoded,
             protocol_info,
-        }
+        })
     }
 
     async fn init_blocks_db(&self) -> Result<BlockDbImpl, MmError<ZcoinClientInitError>> {
-        let cache_db_path = self.db_dir_path.join(format!("{}_cache.db", self.ticker));
-        let ctx = self.ctx.clone();
+        let ctx = &self.ctx;
         let ticker = self.ticker.to_string();
 
-        BlockDbImpl::new(&ctx, ticker, cache_db_path)
-            .map_err(|err| MmError::new(ZcoinClientInitError::ZcoinStorageError(err.to_string())))
-            .await
+        #[cfg(not(target_arch = "wasm32"))]
+        let blocks_db = {
+            let cache_db_path = self.ctx.global_dir().join(format!("{}_cache.db", self.ticker));
```

Comment:
For zcoin we store tx cache in the global dir, so I assume accessible for all wallets on one node.
For other coins, we are going to store TX_CACHE in 'address' dirs.
Do we not want to reconsider doing dirs identically?

Reply:
> For other coins, we are going to store TX_CACHE in 'address' dirs.
> Do we not want to reconsider doing dirs identically?

agreed. will put them in global dir for other coins too as that's easier to manage.

Reply:
done here [28dc4ea](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2398/commits/28dc4eafa283dadec46be9194d1b123ae6964101). thanks, that's simpler.